lib/experimental/media/load.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Adds media-related settings to the block editor.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Returns a list of all available image sizes.
+ *
+ * @since X.X.X
+ *
+ * @return array Existing image sizes.
+ */
+function gutenberg_get_all_image_sizes(): array {
+	$sizes = wp_get_registered_image_subsizes();
+
+	foreach ( $sizes as $name => &$size ) {
+		$size['height'] = (int) $size['height'];
+		$size['width']  = (int) $size['width'];
+		$size['name']   = $name;
+	}
+	unset( $size );
+
+	return $sizes;
+}
+
+/**
+ * Adds media-related settings to the block editor.
+ *
+ * @since X.X.X
+ *
+ * @param array $settings Existing block editor settings.
+ *
+ * @return array New block editor settings.
+ */
+function gutenberg_block_editor_settings_media_processing( $settings ) {
+	if ( ! gutenberg_is_experiment_enabled( 'gutenberg-media-processing' ) ) {
+		return $settings;
+	}
+
+	/** This filter is documented in wp-admin/includes/images.php */
+	$image_size_threshold = (int) apply_filters( 'big_image_size_threshold', 2560, array( 0, 0 ), '', 0 ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+	$settings['__experimentalAvailableImageSizes']   = gutenberg_get_all_image_sizes();
+	$settings['__experimentalBigImageSizeThreshold'] = $image_size_threshold;
```

Comment:
Can you expand a bit on these settings?

I have two thoughs:

 - We're trying to move away slowly from block editor settings passed from the server to the client and instead relying more on the endpoints (The base endpoint has been used for similar things)

Reply:
I can look into moving those to the REST API endpoint.